index.bs

```diff
@@ -9145,48 +9145,54 @@ attributes must begin with a unique identifier for the vendor and user-agent
-entries. Each key must be a member of the [=table of standard storage partition
-key attributes=] or an [=extension storage partition key attribute=].
+key attributes</dfn> which contains zero or more entries. Each key must be a
+member of the [=table of standard storage partition key attributes=] where the
+[=storage partition key=] corresponds to a standard storage partition, or an
+[=extension storage partition key attribute=] where it does not. The precise
+entries are implementation-defined and are determined by the storage
+partitioning adopted by the implementation.
 
 Note: Default values are used by [=storage.getCookies=], [=storage.setCookie=],
 and [=storage.deleteCookies=] when the [=local end=] does not specify a value.
 
 A [=remote end=] has a [=/list=] of <dfn>required partition key
-attributes</dfn> which contains zero or more implementation-defined entries.
-Each key must be a member of the [=table of standard storage partition key
-attributes=] or an [=extension storage partition key attribute=].
+attributes</dfn> which contains zero or more entries. Each key must be a member
+of the [=table of standard storage partition key attributes=] where the
+[=storage partition key=] corresponds to a standard storage partition, or an
+[=extension storage partition key attribute=] where it does not. The precise
+entries are implementation-defined and are determined by the storage
+partitioning adopted by the implementation.
 
 <div algorithm>
 To <dfn>expand a storage partition spec</dfn> given |partition spec|:
 
 1. If |partition spec| is null:
 
-  1. Let |partition key| be an empty [=/map=].
+  1. Set |partition spec| to an empty [=/map=].
 
 1. Otherwise, if |partition spec|["<code>type</code>"] is "<code>context</code>":
 
-  1. Let |browsing context| be the result of [=trying=] to [=get a browsing context=] given |partition spec|.
+  1. Let |browsing context| be the result of [=trying=] to [=get a browsing context=] given |partition spec|["<code>context</code>"].
 
   1. Let |partition key| be the [=storage partition key|key=] of |browsing context|'s [=associated storage partition=].
 
   1. Return [=success=] with data |partition key|.
 
-1. Otherwise:
-
-  1. Assert: |partition spec|["<code>type</code>"] is "<code>storageKey</code>".
-
-  1. Let |partition key| be |partition spec|.
+1. Let |partition key| be an empty [=/map=].
 
 1. For each |name| → |default value| in the [=default values for storage partition key attributes=]:
 
-  1. If |partition key|[|name|] does not [=map/exist=]:
+  1. Let |value| be |partition spec|[|name|] if it [=map/exists=] or |default value| otherwise.
 
-    1. [=map/Set=] |partition key|[|name|] to |default value|.
+  1. [=map/Set=] |partition key|[|name|] to |value|.
 
 1. For each |name| in the remote end's [=required partition key attributes=]:
 
-  1. If |partition key|[|name|] does not [=map/exist=]:
+  1. If |partition spec|[|name|] [=map/exists=]:
```

Comment:
So this means that you're treating "required" and "default" as mutually exclusive? i.e. the full key will be all the required keys plus all the keys with a default value? I was imagining that "required" would be everything and "default" would be the subset for which a known default exists. Either way works, but I found this quite surprising, so we at least need to be clearer in the definitions what's expected.